scripts/compare_default_params.sh

```diff
@@ -1,16 +1,68 @@
 #!/bin/bash
 
-# This script compares default parameters specified in main.star, kurtosis.yml, and params.yml.
-# The true reference for default parameters is params.yml.
+# This script compares default parameters specified in three files: `input_parser.star`,
+# `kurtosis.yml` and `params.yml` where `params.yml` is the source of truth for default parameters.
+# The script outputs the differences between the specified parameters in each file compared to `params.yml`.
 
-echo "Dumping default parameters..."
-sed -n '/DEFAULT_ARGS = {/,/}/ { s/DEFAULT_ARGS = //; s/}/}/; p; }' input_parser.star | yq --yaml-output > input-parser-args.yml
+# Extracting default parameters from the different files.
+echo "Extracting default parameters from input_parser.star..."
+if ! sed -n '/DEFAULT_ARGS = {/,/}/ { s/DEFAULT_ARGS = //; s/}/}/; p; }' input_parser.star | yq --yaml-output >.input_parser.star; then
+  echo "Error: Failed to extract parameters from input_parser.star."
+  exit 1
+fi
+
+echo "Extracting default parameters from kurtosis.yml..."
 # shellcheck disable=SC2016
-sed -n '/```yml/,/```/ { /```yml/d; /```/d; p;}' kurtosis.yml | yq --yaml-output > kurtosis-args.yml
-yq --yaml-output .args params.yml > params-args.yml
+if ! sed -n '/```yml/,/```/ { /```yml/d; /```/d; p;}' kurtosis.yml | yq --yaml-output >.kurtosis.yml; then
+  echo "Error: Failed to extract parameters from kurtosis.yml."
+  exit 1
+fi
+
+echo "Extracting default parameters from params.yml..."
+if ! yq --yaml-output .args params.yml >.params.yml; then
+  echo "Error: Failed to extract parameters from params.yml."
+  exit 1
+fi
+
+# Function to compare files and output differences in a structured format
+# Function to compare files and output differences in a structured format
```

Comment:
nit dupe comment

Reply:
fixed in [chore: clean up](https://github.com/0xPolygon/kurtosis-cdk/pull/218/commits/e6c583851f089e67bd68c5ac8535b482ebe3bd18)